core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,287 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.hello;
+
+import com.google.common.base.Throwables;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.client.WorkflowStub;
+import io.temporal.client.WorkflowUpdateException;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.UpdateMethod;
+import io.temporal.workflow.UpdateValidatorMethod;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample Temporal workflow that demonstrates how to use workflow update methods to update a
+ * workflow execution from external sources.
+ *
+ * <p>Note: Make sure to set {@code frontend.enableUpdateWorkflowExecution=true} in your Temporal
+ * config to enabled update.
+ */
```

Comment:
I don't like this example. It doesn't demonstrate why an update should is better than a signal. 
I would make it simpler to remove the queue and update some workflow state, rejecting it if it was updated from a different client. Some version value can drive the rejection.

Reply:
The update handle does modify workflow state. I think it also does show the benefit of update over a query or a signal because it modifies the workflow state like a signal and it returns a result like a query. That being said it is probably  worth calling out these benefits explicitly here as  well.